MSVC_2022 build with unicode get error:

error message：
1>F:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.36.32532\include\expected(12): warning STL4038: The contents of <expected> are available only with C++23 or later.
1>E:\glaze-1.6.0\include\glaze/api/lib.hpp(109,36): error C2664: “HMODULE LoadLibraryW(LPCWSTR)”: 无法将参数 1 从“const _Elem *”转换为“LPCWSTR”
1>        with
1>        [
1>            _Elem=char
1>        ]
1>E:\glaze-1.6.0\include\glaze/api/lib.hpp(109,58): message : 指向的类型不相关; 转换需要 reinterpret_cast、C 样式强制转换或带圆括号的函数样式强制转换
1>C:\Program Files (x86)\Windows Kits\10\Include\10.0.22000.0\um\libloaderapi.h(601,1): message : 参见“LoadLibraryW”的声明
1>E:\glaze-1.6.0\include\glaze/api/lib.hpp(109,36): message : 尝试匹配参数列表“(const _Elem *)”时
1>        with
1>        [
1>            _Elem=char
1>        ]
1>已完成生成项目“lib_test.vcxproj”的操作 - 失败

add my fix code:

#include <locale>
#include <codecvt>

private:
      bool load_lib(const std::string& path) noexcept
      {
#ifdef GLAZE_API_ON_WINDOWS
          #ifdef UNICODE
                std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> converter;
                lib_t loaded_lib = LoadLibrary(converter.from_bytes(path).c_str());
         #else
                lib_t loaded_lib = LoadLibrary(path.c_str());
         #endif
#else
     .......
    }

--- a/include/glaze/api/lib.hpp
+++ b/include/glaze/api/lib.hpp
@@ -99,8 +99,9 @@ namespace glz
      private:
       bool load_lib(const std::string& path) noexcept
       {
-#ifdef GLAZE_API_ON_WINDOWS
-         lib_t loaded_lib = LoadLibrary(path.c_str());
+#ifdef GLAZE_API_ON_WINDOWS         
+         std::filesystem::path file_path(path);
+         lib_t loaded_lib = LoadLibraryW(file_path.native().c_str());
 #else
          lib_t loaded_lib = dlopen(path.c_str(), RTLD_LAZY);
 #endif

